Experiments/forest_QKS/qks.py

```diff
@@ -0,0 +1,223 @@
+        return square
+
+    def make_barriers(num, l):
+        dist = 0.1*l
+        left = np.random.uniform((-l-dist),(-l+dist),num)
+        right = np.random.uniform((l-dist),(l+dist),num)
+        left_barrier = np.random.uniform(-l,l,num)
+        right_barrier = np.random.uniform(-l,l,num)
+        L = zip_arrays(left,left_barrier)
+        R = zip_arrays(right,right_barrier)
+        barriers = np.vstack((L,R))
+        return barriers
+    
+    outer = make_squares(num, l=2)
+    inner = make_squares(num, l=1)
+    frames = np.concatenate((outer,inner))
+    
+    half = int((len(frames)/2))
+    target = np.concatenate((np.zeros(half),np.ones(half)))
+    
+    X,y = shuffle(frames, target, random_state=1337)
+    return X,y
+
+
+def make_omega(n_episodes,p,q,r,scale,loc=0):
+    """
+    encodes the input vector into q gate parameters
+    """
+
+    def create_selection_matrix(p,q,r):
+        """
+        Generates a matrix of 0s and 1s to zero out `r` values per matrix
+        """
+        matrix_size = p*q
+        m = np.zeros(matrix_size)
+        for i in range(r): m[i] = 1
+        np.random.shuffle(m)
+        selection_matrix = m.reshape(q,p)
+        return selection_matrix
+
+    norm_dist = np.random.normal(loc, scale, size=(n_episodes,q,p))
+    selection_matrix = np.array([create_selection_matrix(p,q,r) for x in range(n_episodes)])
+    omega = norm_dist * selection_matrix # matrix chooses which values to keep
+    return omega
+
+
+def get_beta(n_episodes,q):
+    """
+    q-dimensional bias vector
+    """
+    return np.random.uniform(low=0,high=(2*np.pi),size=(n_episodes,q))
+
+
+def get_theta(omega, u, beta, n_episodes):
+    """
+    random parameters
+    """
+    theta = [omega[e].dot(u[i].T) + beta[e]
+             for i in range(u.shape[0]) 
+             for e in range(n_episodes)
+             ]
+    return np.array(theta)
+
+
+def make_qc(q, noisy=False):
+    q_str = str(q)+'q'
+    qvm = '-qvm'
+    name = q_str+qvm if not noisy else q_str+'-noisy'+qvm
+    print("name of qc: ", name)
+    qc = get_qc(name)
+    return qc
+
+
+def ansatz(qc):
+    """
+    This function creates the quantum circuit and performs the measurements.
+    It mimics the circuits from the appendix of the paper, with the exception of
+    the ordering of the CNOT gates (before and after compilation they still do not match).
+    This is probably best just left up to the compiler.
+
+    For ease of reading the printed operations, the qubits are looped over several times.
+    """    
+    program = Program()
+    qubits = qc.qubits()
+    n_qubits = len(qubits)
+    var = 'theta'
+
+    ro = program.declare('ro', memory_type='BIT', memory_size=n_qubits)
+    thetas = {var+str(qubit):program.declare(var+str(qubit), memory_type = 'REAL') for qubit in qubits}
+
+    sq = int(np.sqrt(n_qubits))
+    lim = n_qubits-sq-1 
+    
+    for m in qubits: program += (RX(thetas[var+str(m)],m))
```

Comment:
I understand it's a one-line for loop syntax.
Please change it to two lines - it's much easier to read then.